### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by NazarLytvyniuk**

### CODE REVIEW

File: app/test_main.py [hunk 1]
```diff
@@ -1,3 +1,26 @@
...
+    [
+        (0, 0, [0, 0]),
+        (14, 14, [0, 0]),
+        (15, 15, [1, 1]),
+        (23, 23, [1, 1]),
+        (24, 24, [2, 2]),
+        (27, 27, [2, 2]),
+        (28, 28, [3, 2]),
+        (29, 29, [3, 3]),
+        (100, 100, [21, 17])
```
Comment (Dimosphen1) [inline app/test_main.py:17]: Consider adding test cases to cover input with negative values, different values for cat_age and dog_age, and to test that the correct exception is raised in case of incorrect data type input.

File: app/test_main.py [hunk 2]
```diff
@@ -1,3 +1,26 @@
...
+    ]
+)
+def test_should_return_correct_ages(
+        cat_age: int,
+        dog_age: int,
+        expected_human_ages: list[int]
+) -> None:
+
```
Comment (Dimosphen1) [inline app/test_main.py:25]: Exceptions processing may include changes in the `test_should_return_correct_ages` function as well.

**Review by Dimosphen1 — CHANGES_REQUESTED** [review]
Additional test cases requested

File: app/test_main.py [hunk 3]
```diff
@@ -1,3 +1,49 @@
...
+    assert get_human_age(cat_age, dog_age) == expected_human_ages
+
+
+@pytest.mark.parametrize(
+    "cat_age, dog_age",
+    [
+        ([], 12),
```
Comment (Dimosphen1) [inline app/test_main.py:36]: Parametrize should include the expected value as well. For this test it would be `exception_type`, rewrite this test to use the following signature "cat_age,dog_age,exception_type".

File: app/test_main.py [hunk 4]
```diff
@@ -1,3 +1,49 @@
...
+        (6, []),
+        (5, "dog")
+    ]
+)
+def test_should_raise_type_error(
+        cat_age: Any,
+        dog_age: Any
+) -> None:
+
+    with pytest.raises(Exception) as error:
```
Comment (Dimosphen1) [inline app/test_main.py:47]: Test that test raises exactly the exception that is needed, not a general exception.

File: app/test_main.py [hunk 5]
```diff
@@ -1,3 +1,49 @@
...
+        get_human_age(cat_age, dog_age)
+    assert error.type == TypeError
```
Comment (Dimosphen1) [inline app/test_main.py:49]: Avoid using `assert` for this test, all work can be done in a context manager.

**Review by Dimosphen1 — CHANGES_REQUESTED** [review]
Several changes to `test_should_raise_type_error` were requested.

**Review by Polyakiv-Andrey — APPROVED** [review]
Good job!
